### PR TITLE
Support oracle alter partition grammar

### LINKED ISSUE #9694: oracle no viable alternative at input 'DROPPARTITION'
## Bug Report

**For English only**, other languages will not accept.

Before report a bug, make sure you have:

- Searched open and closed [GitHub issues](https://github.com/apache/shardingsphere/issues).
- Read documentation: [ShardingSphere Doc](https://shardingsphere.apache.org/document/current/en/overview).

Please pay attention on issues you submitted, because we maybe need more details. 
If no response anymore and we cannot reproduce it on current information, we will **close it**.

Please answer these questions before submitting your issue. Thanks!

Which version of ShardingSphere did you use?
5.0.0-alpha
Which project did you use? ShardingSphere-JDBC or ShardingSphere-Proxy?
ShardingSphere-JDBC
### Expected behavior
when i excute the sql, ALTER TABLE tablename DROP PARTITION partitionname, some wrong message like this:
no viable alternative at input 'DROPPARTITION'
but i exute the sql, ALTER TABLE tablename ADD PARTITION partitionname, that is ok.

before , when i used 4.1.0, this sql can run ok ,ALTER TABLE tablename DROP PARTITION partitionname

### Actual behavior
wrong message:
no viable alternative at input 'DROPPARTITION'

org.antlr.v4.runtime.misc.ParseCancellationException: null

### PR DESCRIPTION
**Pull request by zhujunxxxxx**
Fixes #9694.

Changes proposed in this pull request:
- support oracle alter table add partition
- support oracle alter table drop partition

depend on: `https://docs.oracle.com/en/database/oracle/oracle-database/18/sqlrf/ALTER-TABLE.html#GUID-552E7373-BF93-477D-9DA3-B2C9386F2877`

### CODE REVIEW

**Review by tristaZero — CHANGES_REQUESTED** [review]
Hi @zhujunxxxxx LGTM.
@wgy8283335 , could you give a review here? I am unsure your previous work covered this one or not.

**Comment by zhujunxxxxx** [comment]
> Hi @zhujunxxxxx LGTM.
> @wgy8283335 , could you give a review here? I am unsure your previous work covered this one or not.

this PR just support SS can execute the SQL like `alter table t add partition .... ` and `alter table t drop partition ....`. 
The question posed by @wgy8283335 is not of the same dimension as issues #9694

File: shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-oracle/src/main/antlr4/imports/oracle/DDLStatement.g4 [hunk 1]
```diff
@@ -919,3 +919,31 @@ flashbackArchiveClause
...
+    ((LP_? rangeSubpartitionDesc (COMMA_ rangeSubpartitionDesc)* | listSubpartitionDesc (COMMA_ listSubpartitionDesc)* | individualHashSubparts (COMMA_ individualHashSubparts)* RP_?)
+        | hashSubpartitionQuantity)?
+    ;
+
+addListPartitionClause
+    : PARTITION partitionName? listValuesClause tablePartitionDescription
+    ((LP_? rangeSubpartitionDesc (COMMA_ rangeSubpartitionDesc)* | listSubpartitionDesc (COMMA_ listSubpartitionDesc)* | individualHashSubparts (COMMA_ individualHashSubparts)* RP_?)
+    | hashSubpartitionQuantity)?
+    ;
+
```
Comment (wgy8283335) [inline shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-oracle/src/main/antlr4/imports/oracle/DDLStatement.g4:942]: The hash partition and system partition also need to be added.

Comment (wgy8283335) [reply shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-oracle/src/main/antlr4/imports/oracle/DDLStatement.g4:942]: The addRangePartitionClause and addListPartitionClause are not fully completed, below rules need to be done.
table_partition_description
external_part_subpart_data_props
update_index_clauses

Comment (zhujunxxxxx) [reply shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-oracle/src/main/antlr4/imports/oracle/DDLStatement.g4:942]: @wgy8283335 
I know, but I just used the existing syntax to build it. If want to support fully completed grammer maybe need @tristaZero open a issues.
By the way, the grammer of partition in `CREATE TABLE` are also not support  fully completed grammer.

Comment (tristaZero) [reply shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-oracle/src/main/antlr4/imports/oracle/DDLStatement.g4:942]: Hi, We can have a plan to support this grammar completely. Firstly, we can add the whole grammar definition, then we can consider improve the class of `CreateTableStatement`.

Comment (zhujunxxxxx) [reply shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-oracle/src/main/antlr4/imports/oracle/DDLStatement.g4:942]: @tristaZero agree, this PR can merge first?

Comment (tristaZero) [reply shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-oracle/src/main/antlr4/imports/oracle/DDLStatement.g4:942]: So 
```
The addRangePartitionClause and addListPartitionClause are not fully completed, below rules need to be done.
table_partition_description
external_part_subpart_data_props
update_index_clauses
```
Will these definitions be improved in the next PR?
It is not a complete grammar definition, is it?

Comment (zhujunxxxxx) [reply shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-oracle/src/main/antlr4/imports/oracle/DDLStatement.g4:942]: Next PR complete the grammer of partition in `CreateTableStatement` and `AlterTableStatement` .

Comment (tristaZero) [reply shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-oracle/src/main/antlr4/imports/oracle/DDLStatement.g4:942]: Great.

**Review by tristaZero — APPROVED** [review]
Great, just do what you want step by step.

**Pull request merged**